src/analyzer/protocol/ldap/ldap.spicy

```diff
@@ -11,625 +11,612 @@ import spicy;
+    var responseToken: optional<bytes>;
+
+    # Parse the contained Sequence.
+    seq: ASN1::ASN1Message(True) {
+        for (msg in $$.body.seq.submessages) {
+            # https://www.rfc-editor.org/rfc/rfc4178#section-4.2.2
+            if (msg.application_id == 0) {
+                self.accepted = msg.application_data == b"\x0a\x01\x00";
+            } else if (msg.application_id == 1) {
+                self.supportedMech = msg;
+            } else if (msg.application_id == 2) {
+                self.responseToken = msg.application_data;
+            } else if (msg.application_id == 3) {
+                # ignore mechListMec
+            } else {
+                throw "unhandled NegTokenResp id %s" % msg.application_id;
+            }
+        }
     }
-  }
 
-  switch ( self?.supportedMech ) {
-    True -> supportedMechOid: ASN1::ASN1Message(False) &convert=$$.body.str_value;
-    * -> : void;
-  } &parse-from=self.supportedMech.application_data;
+    switch (self?.supportedMech) {
+        True -> supportedMechOid: ASN1::ASN1Message(False) &convert=$$.body.str_value;
+        * -> : void;
+    } &parse-from=self.supportedMech.application_data;
 };
 
 # TODO(fox-ds): A helper unit for requests for which no handling has been implemented.
 # Eventually all uses of this unit should be replaced with actual parsers so this unit can be removed.
 type NotImplemented = unit(inout message: Message) {
-  : skip bytes &eod;
+    : skip bytes &eod;
 };
 
 type BindRequest = unit(inout message: Message, ctx: Ctx&) {
-  version: ASN1::ASN1Message(True) &convert=$$.body.num_value;
-  name: ASN1::ASN1Message(True) &convert=$$.body.str_value {
-    message.obj = self.name;
-  }
-  var authType: BindAuthType = BindAuthType::Undef;
-  var authData: bytes = b"";
-  var simpleCreds: string = "";
-
-  : ASN1::ASN1Message(True) {
-    if ($$?.application_id) {
-      self.authType = cast<BindAuthType>(cast<uint8>($$.application_id));
-      self.authData = $$.application_data;
+    version: ASN1::ASN1Message(True) &convert=$$.body.num_value;
+    name: ASN1::ASN1Message(True) &convert=$$.body.str_value {
+        message.obj = self.name;
     }
-  }
-
-  if ( |self.authData| > 0 ) {
-    switch ( self.authType ) {
-      BindAuthType::BIND_AUTH_SIMPLE ->
-        : void {
-          self.simpleCreds = self.authData.decode();
-          message.arg = self.simpleCreds;
+    var authType: BindAuthType = BindAuthType::Undef;
+    var authData: bytes = b"";
+    var simpleCreds: string = "";
+
+    : ASN1::ASN1Message(True) {
+        if ($$?.application_id) {
+            self.authType = cast<BindAuthType>(cast<uint8>($$.application_id));
+            self.authData = $$.application_data;
         }
+    }
 
-      BindAuthType::BIND_AUTH_SASL ->
-        saslCreds: SaslCredentials {
-          message.arg = self.saslCreds.mechanism;
-          ctx.saslMechanism = self.saslCreds.mechanism;
+    if (|self.authData| > 0) {switch (self.authType) {
```

Comment:
I released spicy-format-0.26.0 to insert a line break after the `if .. {`.

Reply:
Nice catch! Thanks!